src/Microsoft.AspNet.Owin/OwinExtensions.cs

```diff
@@ -67,21 +68,32 @@ public static IApplicationBuilder UseOwin(this IApplicationBuilder builder, Acti
 
         public static IApplicationBuilder UseBuilder(this AddMiddleware app)
         {
+            return app.UseBuilder(configureServices: null);
+        }
+
+        public static IApplicationBuilder UseBuilder(this AddMiddleware app, Action<IServiceCollection> configureServices)
```

Comment:
I'm not understanding your concern, what do you want to configure exactly? There is the `ApplicationServices` property on the `IApplicationBuilder` from which you can access the application's service provider. The intended use of these extensions are illustrated here https://github.com/aspnet/Entropy/pull/72